crates/engine/tree/src/tree/mod.rs

```diff
@@ -2176,14 +2184,115 @@ where
             return Err(e.into())
         }
 
+        let code_cache = Cache::new(1000000);
+        let storage_cache = Cache::new(1000000);
+        let account_cache = Cache::new(1000000);
+        let metrics = CachedStateMetrics::default();
+
         trace!(target: "engine::tree", block=?block.num_hash(), "Executing block");
-        let executor = self.executor_provider.executor(StateProviderDatabase::new(&state_provider));
+        let cached_state_provider = CachedStateProvider::new(
+            &state_provider,
+            code_cache.clone(),
+            storage_cache.clone(),
+            account_cache.clone(),
+            metrics.clone(),
+        );
+        let executor =
+            self.executor_provider.executor(StateProviderDatabase::new(&cached_state_provider));
 
         let block_number = block.number;
         let block_hash = block.hash();
         let sealed_block = Arc::new(block.block.clone());
         let block = block.unseal();
 
+        info!(target: "engine::tree", "Starting account, code, storage prewarm");
+        let txs_with_senders: Vec<_> = block.transactions_with_sender().collect();
+
+        // accumulate all senders and access lists for an explicit warm, that does not execute.
+        let cloned_code_cache = code_cache.clone();
+        let cloned_storage_cache = storage_cache.clone();
+        let cloned_account_cache = account_cache.clone();
+        let cloned_metrics = metrics.clone();
+        let shared_state_provider = self.state_provider(block.parent_hash)?.unwrap();
+        let to_addrs = txs_with_senders.iter().filter_map(|item| item.1.to());
+        let senders: Vec<_> = txs_with_senders.iter().map(|item| *item.0).chain(to_addrs).collect();
+        rayon::spawn(move || {
```

Comment:
I think this should not use rayon because very IO heavy

Reply:
yes, this is just abusing its global threadpool, we will need to use a different threadpool